### PR TITLE
raise 403 instead of 500 on invalid Alice Signature on kfrag

### PR DESCRIPTION
**Pull request by vepkenez**
**Type of PR:**
- [x] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
500s will not be raised in response to HTTP requests.
Also, 400s all have sensible messages.

**Why it's needed:**
Better http behavior.

### CODE REVIEW

**Comment by KPrasch** [comment]
This PR needs a rebase over master for CI to pass.

File: nucypher/network/server.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def set_policy(id_as_hex):
         kfrag = KFrag.from_bytes(kfrag_bytes)
 
         if not kfrag.verify(signing_pubkey=alices_verifying_key):
-            raise InvalidSignature("{} is invalid".format(kfrag))
+            return Response(f"Signature on {kfrag} is invalid", status=403)
```
Comment (derekpierre) [inline nucypher/network/server.py:261]: Should we also do something similar on L265 where we raise `alice.SuspiciousActivity`?

Comment (vepkenez) [reply nucypher/network/server.py:261]: Yes that’s a good idea.  Will add it now.

File: nucypher/network/server.py [hunk 2]
```diff
@@ -246,23 +246,23 @@ def set_policy(id_as_hex):
...
 
         with datastore.describe(PolicyArrangement, id_as_hex, writeable=True) as policy_arrangement:
             if not policy_arrangement.alice_verifying_key == alice.stamp.as_umbral_pubkey():
-                raise alice.SuspiciousActivity
+                return Response(f"Policy key does not match Alice stamp.", status=403)
```
Comment (derekpierre) [inline nucypher/network/server.py:265]: Tricky wording here. Perhaps something like:

```suggestion
                return Response("Policy arrangement's signing key does not match sender's", status=403)
```

File: nucypher/network/server.py [hunk 3]
```diff
@@ -246,23 +246,23 @@ def set_policy(id_as_hex):
...
                 # Alice didn't pay.  Return response with that weird status code.
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"No transaction matching {tx}."))
-                return Response(status=402)
+                return Response(f"No paid transaction matching {tx} for this node", status=402)
 
             this_node_has_been_arranged = this_node.checksum_address in arranged_addresses
             if not this_node_has_been_arranged:
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"The transaction {tx} does not list me as a Worker - it lists {arranged_addresses}."))
-                return Response(status=402)
+                return Response(f"No transaction arrangement matching {tx} for this node", status=402)
```
Comment (derekpierre) [inline nucypher/network/server.py:254]: wdyt?

```suggestion
                return Response(f"This node was not listed as servicing the policy in transaction {tx}", status=402)
```

**Review by derekpierre — APPROVED** [review]
🎸 - thanks for updating the other spots without messages.

A couple suggestions - let me know what you think.

**Comment by derekpierre** [comment]
Oh, perhaps we can add a newsfragment as well since we're changing the status code.

**Comment by vepkenez** [comment]
> Oh, perhaps we can add a newsfragment as well since we're changing the status code.

yeah yeah 🙄 😆

File: nucypher/network/server.py [hunk 4]
```diff
@@ -246,23 +246,23 @@ def set_policy(id_as_hex):
...
                 # Alice didn't pay.  Return response with that weird status code.
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"No transaction matching {tx}."))
-                return Response(status=402)
+                return Response(f"No paid transaction matching {tx} for this node", status=402)
 
             this_node_has_been_arranged = this_node.checksum_address in arranged_addresses
             if not this_node_has_been_arranged:
                 this_node.suspicious_activities_witnessed['freeriders'].append((alice, f"The transaction {tx} does not list me as a Worker - it lists {arranged_addresses}."))
-                return Response(status=402)
+                return Response(f"This node was not listed as servicing the policy in transaction {tx}", status=402)
```
Comment (KPrasch) [inline nucypher/network/server.py:254]: Gotta love the usage of 402 here.  I've waited a lifetime to use it.

> 402 Payment Required 
   Reserved for future use. The original intention was that this code might be used as part of some form of digital cash or micropayment scheme

**Review by KPrasch — APPROVED** [review]
:cowboy_hat_face:

**Pull request merged**